lib/optimize/SplitChunksPlugin.js

```diff
@@ -85,7 +87,15 @@ module.exports = class SplitChunksPlugin {
 				const names = chunks.map(c => c.name);
 				if(!names.every(Boolean)) return;
 				names.sort();
-				const name = (cacheGroup && cacheGroup !== "default" ? cacheGroup + "~" : "") + names.join("~");
+				let name = (cacheGroup && cacheGroup !== "default" ? cacheGroup + "~" : "") + names.join("~");
+				// Filename length can't exceed 255 characters otherwise an
+				// ENAMETOOLONG error is raised. If the generated name if too
+				// long, it is truncated and a unique ID is append. The limit
+				// has been set to 200 to prevent `[name].[chunkhash].[ext]`
+				// from generating a 256+ character string.
+				if(name.length > 200) {
+					name = name.slice(0, 200) + "~" + (++guardID);
```

Comment:
The guardID is not a good idea.

This method is called multiple times for the same combination of the chunk (for each module) and should return the same result.

In addition to that this depends on execution timing and makes builds non-deterministic.

Just return `undefined` in this case, which give the chunk no name, which is fine. Chunks don't need names, it's just for cooler filenames.

Reply:
Alternative: truncate it and add a hash of the full name, that's deterministic, but not sure how useful this would be.